pollbot/telegram/message_handler.py

```diff
@@ -249,3 +256,38 @@ def handle_user_option_addition(
         update_poll_messages(session, bot, poll)
     else:
         chat.send_message(i18n.t("creation.option.no_new", locale=user.locale))
+
+
+def handle_due_time_duration(
+    bot: Bot,
+    update: Update,
+    session: scoped_session,
+    user: User,
+    text: str,
+    poll: Poll,
+    chat: Chat,
+) -> Optional[str]:
+    """Set the due time duration of the poll."""
+
+    if not text.isnumeric():
+        return i18n.t("creation.error.limit_bigger_zero", locale=user.locale)
+
+    "Set 1 week in minutes as arbitrary maximum relative value"
+    maximum_relative_minutes = 60 * 24 * 7
+    relative_minutes = int(text)
+    if relative_minutes < 1 or relative_minutes > maximum_relative_minutes:
+        return i18n.t(
+            "creation.error.limit_between",
+            locale=user.locale,
+            limit=maximum_relative_minutes,
+        )
+
+    poll_due_date = datetime.now()
+    next_poll_due_date = poll_due_date + relativedelta(minutes=relative_minutes)
+    poll.set_due_date(next_poll_due_date)
+
+    chat.send_message(
+        text=get_due_time_duration_text(poll),
+        reply_markup=get_add_due_time_keyboard(poll),
+        parse_mode="markdown",
+    )
```

Comment:
To be honest, I didn't expect this to be a relative number input in minutes.
This is super hard to properly convey to users, especially to users that aren't used to work with time that way.

I imagined this to be built into the already existing keyboard.
Something like

```
DATEPICKER HERE
[       Hours         ]
[+4][+1][-1][-4]
[       Minutes      ]
[+15][+5][+1][-1][-5][-15]
```